nodeup/pkg/model/kubelet.go

```diff
@@ -589,6 +589,9 @@ func (b *KubeletBuilder) buildKubeletConfigSpec() (*kops.KubeletConfigSpec, erro
 		if len(c.Taints) == 0 && isMaster {
 			// (Even though the value is empty, we still expect <Key>=<Value>:<Effect>)
 			c.Taints = append(c.Taints, nodelabels.RoleLabelMaster16+"=:"+string(v1.TaintEffectNoSchedule))
+			if b.IsKubernetesGTE("1.24") {
+				c.Taints = append(c.Taints, nodelabels.RoleLabelControlPlane20+"=:"+string(v1.TaintEffectNoSchedule))
```

Comment:
It's probably a good thing to get this into testing, but we could also split this PR up to _not_ add the new taints / drop the old labels _yet_.  Ideally we would get this into 1.24, but if users have problems with it we might expose a feature flag..  That said, I think you've shown that it isn't too hard to write a compatible manifest.

Reply:
I don't think the migration will be easier than it is right now. And we've both warned about this for some time and kubeadm is now making the move.